libs/wingsdk/src/http/http.ts

```diff
@@ -0,0 +1,267 @@
+   * * If the server indicates that the resource has not changed, it will be returned from the cache. Otherwise the resource will be downloaded from the server and the cache will be updated.
+   * * If there is no match, the runtime environment will make a normal request, and will update the cache with the downloaded resource.
+   */
+  NO_CACHE = "no-cache",
+  /**
+   * The runtime environment looks for a matching request in its HTTP cache.
+   * * If there is a match, fresh or stale, it will be returned from the cache.
+   * * If there is no match, the runtime environment will make a normal request, and will update the cache with the downloaded resource.
+   */
+  FORCE_CACHE = "force-cache",
+}
+
+/**
+ * The redirect read-only property that contains the mode for how redirects are handled.
+ */
+export enum RequestRedirect {
+  /**
+   * Follow all redirects incurred when fetching a resource.
+   */
+  FOLLOW = "follow",
+  /**
+   * Return a network error when a request is met with a redirect.
+   */
+  ERROR = "error",
+}
+
+/**
+ * The request's method
+ */
+export enum HttpMethod {
+  /**
+   * GET
+   */
+  GET = "GET",
+  /**
+   * PUT
+   */
+  PUT = "PUT",
+  /**
+   * DELETE
+   */
+  DELETE = "DELETE",
+  /**
+   * PATCH
+   */
+  PATCH = "PATCH",
+  /**
+   * POST
+   */
+  POST = "POST",
+}
+
+/**
+ * An object containing any custom settings that you want to apply to the request.
+ */
+export interface RequestOptions {
+  /**
+   * The request method, e.g., GET, POST. The default is GET.
+   * @default GET
+   */
+  readonly method?: HttpMethod;
+  /**
+   * Any headers you want to add to your request.
+   */
+  readonly headers?: Record<string, string>;
+  /**
+   * Any body that you want to add to your request.
+   * Note that a request using the GET or HEAD method cannot have a body.
+   */
+  readonly body?: string;
+  /**
+   * The cache mode you want to use for the request.
+   */
+  readonly cache?: RequestCache;
+  /**
+   * he redirect mode to use: follow, error. The default is follow.
+   * @default follow
+   */
+  readonly redirect?: RequestRedirect;
+  /**
+   * A string specifying "no-referrer", client, or a URL. The default is "about:client".
+   * @default  about:client
+   */
+  readonly referrer?: string;
+}
+/**
+ * Represents the response to a request.
+ */
+export interface Response {
+  /**
+   * The status code of the response. (This will be 200 for a success).
+   */
+  readonly status: number;
+  /**
+   * The URL of the response.
+   */
+  readonly url: string;
+  /**
+   * A boolean indicating whether the response was successful (status in the range 200 – 299) or not.
+   */
+  readonly ok: boolean;
+  /**
+   * The map of header information associated with the response.
+   */
+  readonly headers: Record<string, string>;
+  /**
+   * A string represents the body contents.
+   */
+  readonly body: string;
```

Comment:
I wonder if this should be optional, to handle the case where some responses may not have a body (for example, HEAD requests)

Reply:
<img width="833" alt="image" src="https://github.com/winglang/wing/assets/39455181/c4c9e9f7-e2c2-45cd-a5df-d809cd2dc59a">